src/environmentbase/template.py

```diff
@@ -594,6 +633,7 @@ def add_asg(self,
 
         if type(instance_type) != str:
             instance_type = Ref(instance_type)
+            raise Exception("Tempalte.add_asg::instance_type should be String")
```

Comment:
Use a custom exception define at the top of the file. Like `TemplateValueError` or so.